modules/lsp.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  config,
+  ...
+}:
+let
+  inherit (lib) mkOption types;
+
+  cfg = config.lsp;
+
+  serverType = types.submodule {
+    options = {
+      enable = lib.mkEnableOption "the language server";
+
+      config = mkOption {
+        type = with types; attrsOf anything;
+        description = ''
+          Configurations for each language server.
+        '';
+        default = { };
+        example = {
+          cmd = [
+            "clangd"
+            "--background-index"
+          ];
+          root_markers = [
+            "compile_commands.json"
+            "compile_flags.txt"
+          ];
+          filetypes = [
+            "c"
+            "cpp"
+          ];
+        };
+      };
+    };
+  };
+in
+{
+  options.lsp = {
+    servers = mkOption {
```

Comment:
Q: what other options do you see being introduced in the `lsp.*` namespace?

Would it ever make sense to rename `lsp.servers` to simply `lsp`?

Reply:
inlayHints, keymaps, etc. could be in there too